toolchains/solidity/linter/core/solidhunter-lib/src/rules/security/no_inline_assembly.rs

```diff
@@ -0,0 +1,76 @@
+use crate::linter::SolidFile;
+use crate::rules::types::*;
+use crate::types::*;
+use ast_extractor::*;
+
+pub const RULE_ID: &str = "no-inline-assembly";
+const MESSAGE: &str = "Avoid to use inline assembly. It is acceptable only in rare cases";
+
+pub struct NoInlineAssembly {
+    data: RuleEntry,
+}
+
+impl NoInlineAssembly {
+    fn create_diag(
+        &self,
+        file: &SolidFile,
+        location: (LineColumn, LineColumn),
+    ) -> LintDiag {
+        LintDiag {
+            id: RULE_ID.to_string(),
+            range: Range {
+                start: Position {
+                    line: location.0.line,
+                    character: location.0.column,
+                },
+                end: Position {
+                    line: location.1.line,
+                    character: location.1.column,
+                },
+            },
+            message: MESSAGE.to_string(),
+            severity: Some(self.data.severity),
+            code: None,
+            source: None,
+            uri: file.path.clone(),
+            source_file_content: file.content.clone(),
+        }
+    }
+}
+
+impl RuleType for NoInlineAssembly {
+    fn diagnose(&self, file: &SolidFile, _files: &[SolidFile]) -> Vec<LintDiag> {
+        let mut res = Vec::new();
+
+        for contract in retriever::retrieve_contract_nodes(&file.data) {
+            for function in retriever::retrieve_functions_nodes(&contract) {
+                if function.body().is_none() {
+                    continue;
+                }
+                let wow = function.body().unwrap();
+                for stmt in wow.iter() {
+                    if let Stmt::Assembly(_) = stmt {
+                        let location = (stmt.span().start(), stmt.span().end());
+                        res.push(self.create_diag(file, location));
+                    }
+                }
```

Comment:
Use the new alloy version (organization repo until it is properly released) to use the stmt_assembly hook